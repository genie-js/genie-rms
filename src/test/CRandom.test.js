const CRandom = require('../CRandom')

test('generator sequence seeded with 0, first ten values', () => {
  const generator = new CRandom(0)
  let values = []
  for (let i = 0; i < 10; i++) {
    values.push(generator.next())
  }
  expect(values).toMatchSnapshot()
})

test('generator range sequence seeded with 0, first ten values', () => {
  const generator = new CRandom(0)
  let values = []
  for (let i = 0; i < 10; i++) {
    values.push(generator.nextRange(10))
  }
  expect(values).toMatchSnapshot()
})
