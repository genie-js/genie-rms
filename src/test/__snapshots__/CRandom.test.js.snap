// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator range sequence seeded with 0, first ten values 1`] = `
Array [
  0,
  2,
  6,
  0,
  2,
  3,
  2,
  9,
  3,
  9,
]
`;

exports[`generator sequence seeded with 0, first ten values 1`] = `
Array [
  38,
  7719,
  21238,
  2437,
  8855,
  11797,
  8365,
  32285,
  10450,
  30612,
]
`;
